src/ui/Facet/FacetSearchDropdownNavigation/DefaultSearchDropdownNavigator.ts

```diff
@@ -0,0 +1,73 @@
+import { first, last } from 'underscore';
+import { $$, Dom } from '../../../utils/Dom';
+
+export interface ISearchDropdownNavigator {
+  nextFocusableElement: () => void;
+  previousFocusableElement: () => void;
+  currentResult: Dom;
+  setAsCurrentResult: (el: Dom) => void;
+}
+
+export interface ISearchDropdownConfig {
+  input: HTMLInputElement;
+  searchResults: HTMLElement;
+  setScrollTrigger: (value: boolean) => void;
+}
+
+export class DefaultSearchDropdownNavigator implements ISearchDropdownNavigator {
+  public currentResult: Dom;
+
+  constructor(private config: ISearchDropdownConfig) {}
+
+  public setAsCurrentResult(toSet: Dom) {
+    this.currentResult && this.currentResult.removeClass('coveo-facet-search-current-result');
+    this.currentResult = toSet;
+    toSet.addClass('coveo-facet-search-current-result');
+    this.updateSelectedOption(toSet);
+  }
+
+  public nextFocusableElement() {
```

Comment:
Could we simply rename `moveCurrentResultDown` / `moveCurrentResultUp` to `nextFocusableElement`  here ?

Reply:
I could, but I prefer to keep this separation because `moveCurrentResultDown/Up`. I find these names more descriptive. Because they are `public`, they convey what is happening when used inside the FacetSearchDropdownNavigator.

If I changed the name, then the logic when be harder to understand on line 30 in `FacetSearchDropdownNavigator.ts`